code/modules/mob/living/carbon/xenomorph/charge_crush.dm

```diff
@@ -370,7 +379,8 @@
 
 /datum/action/xeno_action/ready_charge/bull_charge/on_xeno_upgrade()
 	var/mob/living/carbon/xenomorph/X = owner
-	agile_charge = (X.upgrade == XENO_UPGRADE_FOUR)
+	if(X.upgrade == XENO_UPGRADE_FOUR)
+		agile_charge = 2
```

Comment:
What about going from primo to ancient, like you can do it in testing or admin.

And the previous form was better, should be a trinary